tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -2173,6 +2173,49 @@ public function test_get_post_with_password_without_permission() {
 		$this->assertTrue( $data['excerpt']['protected'] );
 	}
 
+	/**
+	 * @ticket 61837
+	 */
+	public function test_get_item_permissions_check_while_updating_password() {
+		$endpoint = new WP_REST_Posts_Controller( 'post' );
+
+		$request_with_post_password = new WP_REST_Request( 'POST', sprintf( '/wp/v2/posts/%d', self::$post_id ) );
```

Comment:
This will work because of parameter order. But if we want to simulate the request object more closely to how the controller expects it, the `id` should be also listed in `set_url_params`. That's what holds parameters parsed from the actual REST Route.